browser/ui/omnibox/brave_omnibox_client_impl.cc

```diff
@@ -114,3 +109,13 @@ void BraveOmniboxClientImpl::OnTextChanged(
   // match or not when current input is accepted.
   user_text_ = user_text;
 }
+
+void BraveOmniboxClientImpl::OnURLOpenedFromOmnibox(OmniboxLog* log) {
```

Comment:
Oh, using `OnURLOpenedFromOmnibox` seems better.
I think we can delete `OnTextChanged()` overriding and `user_text_` and instead `log->text` can be used.

Reply:
I tried to do this at first. `IsBraveSearchPromotionMatch` breaks because the full suggested search URL would be passed into the function (i.e. `https://search.brave.com/search?q=test`) instead of just the query text.

Reply:
Ok, then the comment might be wrong - `// The user's input text in the omnibox.` :)

Reply:
When navigating to the promo using the keyboard keys, the omnibox input text changes to contain the full URL. So the comment seems correct in that way. `user_text_` allows us to cache the original omnibox text inputted by the user.

Reply:
I see. From omnibox's point of view, its whole text could be user's input.